### DISCUSSION TITLE
Re-run unavaialble for jobs with expired logs

### DISCUSSION THREAD

**Discussion by MattIPv4** [community/community#49242] [Actions]
### Select Topic Area

Bug

### Body

👋 I have some old PRs that have failed jobs on them. The logs for these jobs have now expired, and the UI no longer has a way for me to even re-run the jobs. This has left me rather stuck...

![image](https://user-images.githubusercontent.com/12371363/222973295-4e96f6b8-da84-455b-bddf-f1243a3b100a.png)
![image](https://user-images.githubusercontent.com/12371363/222973305-09262cb2-8584-4963-83d0-b8f17cae4686.png)
![image](https://user-images.githubusercontent.com/12371363/222973315-9fb99830-5be6-4869-9095-7a2527d9fca6.png)

How do I trigger a re-run of this?

**Comment by jsoref** [comment]
As you created the branch and the commits, you should be able to check out the branch, run `git commit --amend --no-edit` and `git push origin HEAD --force-with-lease` to trigger a new run w/ a new datestamp (and thus a new sha). Personally, instead of just doing that, I'd encourage you to also do: `git fetch origin` and `git rebase origin/master` before doing the push (in doing this, you can skip the `git commit --amend --no-edit` as the rebase will have a similar effect).

I understand this is perhaps less than ideal. But it will get you unstuck.

**Reply by MattIPv4** [reply]
Yah, I realised this could be re-triggered by pushing to the branch, but this does rather seem like a bug that the UI offers no way to re-run once logs expire -- the workflow file is still the same and it knows that (clicking on the workflow file tab shows it as expected), it just doesn't let you run it again against the existing commit.

**Reply by jsoref** [reply]
I don't remember how much state a `rerun` uses. (I actually just used rerun which is why I'm reviewing all of the actions discussions because I tripped on a bug while doing it.)

Personally, if I were you, I'd open a ticket to https://support.github.com/ asking for the ability. If you're really lucky they'll add it to the list of things to do, or they might explain why they won't. Either way, don't hold your breath. File the ticket using the most expensive org you have (if you have a GHES/GHEC, use that, if you have a paid Teams, use that, or use your Pro account...) [note: enterprise tickets may rely on a different domain, I'm not a member of one, so I don't have the url for it offhand].

**Reply by jaeyson** [reply]
> As you created the branch and the commits, you should be able to check out the branch, run `git commit --amend --no-edit` and `git push origin HEAD --force-with-lease` to trigger a new run w/ a new datestamp (and thus a new sha). Personally, instead of just doing that, I'd encourage you to also do: `git fetch origin` and `git rebase origin/master` before doing the push (in doing this, you can skip the `git commit --amend --no-edit` as the rebase will have a similar effect).
> 
> I understand this is perhaps less than ideal. But it will get you unstuck.

This is the only solution for now.
